src/BenchmarkDotNet/Running/BenchmarkRunnerDirty.cs

```diff
@@ -73,8 +81,52 @@ public static Summary RunSource(string source, IConfig config = null)
         }
 
         [MethodImpl(MethodImplOptions.NoInlining)]
-        private static Summary RunWithDirtyAssemblyResolveHelper(Type type, IConfig config)
-            => BenchmarkRunnerClean.Run(new[] { BenchmarkConverter.TypeToBenchmarks(type, config) }).Single();
+        private static Summary RunWithDirtyAssemblyResolveHelper(Type type, IConfig config, string[] args)
+        {
+            var logger = config.GetNonNullCompositeLogger();
+            var (isParsingSuccess, parsedConfig, options) = ConfigParser.Parse(args, logger, config);
+            if (!isParsingSuccess) // invalid console args, the ConfigParser printed the error
+                return null;
+
+            if (options.PrintInformation)
+            {
+                logger.WriteLine(HostEnvironmentInfo.GetInformation());
+                return null;
+            }
+
+            var effectiveConfig = ManualConfig.Union(config, parsedConfig);
+
+            types.Add(type);
+            var (allTypesValid, allAvailableTypesWithRunnableBenchmarks) = TypeFilter.GetTypesWithRunnableBenchmarks(types, assemblies, logger);
+            if (!allTypesValid) // there were some invalid and TypeFilter printed errors
+                return null;
+
+            if (allAvailableTypesWithRunnableBenchmarks.IsEmpty())
+            {
+                userInteraction.PrintNoBenchmarksError(logger);
+                return null;
+            }
+
+            if (options.ListBenchmarkCaseMode != ListBenchmarkCaseMode.Disabled)
+            {
+                PrintList(logger, effectiveConfig, allAvailableTypesWithRunnableBenchmarks, options);
+                return null;
+            }
+
+            var benchmarksToFilter = options.UserProvidedFilters
+                ? allAvailableTypesWithRunnableBenchmarks
+                : userInteraction.AskUser(allAvailableTypesWithRunnableBenchmarks, logger);
```

Comment:
if the user did not provide any filters, then you should not ask the user for the choice (to keep the current behavior)

Sth like:

```cs
var benchmarks = options.UserProvidedFilters
   ? TypeFilter.Filter(effectiveConfig, allAvailableTypesWithRunnableBenchmarks)
   : BenchmarkConverter.TypeToBenchmarks(type, config);

BenchmarkRunnerClean.Run(benchmarks).Single();
```

Reply:
this simplify's code a bit.